### DISCUSSION TITLE
how I can install the framework for use in Swift

### DISCUSSION THREAD

**Discussion by ecr334** [ngageoint/simple-features-proj-ios#4] [Q&A]
Please advise in detail how I can install the framework for use in Swift , the steps I have taken:
1) installed pod 'sf-proj-ios', '~> 6.0.3'.
2) created a new header file and inserted #import "sf-proj-ios-Bridging-Header.h" into it.
now it looks like 

```
#ifndef Header_h
#define Header_h

#import "sf-proj-ios-Bridging-Header.h"
```

#endif /* Header_h */
3) I press cmd+b to build the project and 22 errors appear, screenshot attached. 
![Снимок экрана 2024-04-25 в 18 19 03](https://github.com/ngageoint/simple-features-proj-ios/assets/16240094/cab68696-f009-41dd-ba9b-a790b2a0da7a)

**Comment by ecr334** [comment]
if I remove use_framework! 
the project doesn't give these errors anymore, but when I use import sf_proj_ios and try to transform using this transformation
```
guard let coordinateFirst = PROJLocationCoordinate3D(coordinate: CLLocationCoordinate2D(latitude: 48.09763488, longitude: 8.58171633), andZ: 698.458) else { return }
          let proj1 = PROJProjectionFactory.projection(withAuthority: PROJ_AUTHORITY_EPSG, andIntCode: 4326, andParams: "+proj=longlat +datum=WGS84 +no_defs")
         let projEVRF : PROJProjection : PROJProjection = PROJProjectionFactory.projection(withAuthority: PROJ_AUTHORITY_EPSG, andIntCode: 4314, andParams: "+proj=longlat +ellps=bessel +towgs84=598.1,73.7,418.2,0.202,0.045,-2.455,6.7 +no_defs")
        guard let projTRANS = PROJProjectionTransform(from: proj1, andTo: projEVRF) else { return }
        print("proj-Z:", projTRANS.transform3d(coordinateFirst).z)
```

fatal crash.

```
*** Terminating app due to uncaught exception 'Resource Not Found', reason: 'Failed to find resource 'PROJ.bundle/proj' of type 'db''
*** First throw call stack:
(0x1a6cd2b28 0x19eb26f78 0x1a6d582a4 0x100c8bef8 0x100c8bc6c 0x100c8e100 0x19eb04c3c 0x19eb07528 0x19eb05e7c 0x19eb04fe0 0x19eb000c4 0x10082c118 0x10082c848 0x1a90c130c 0x1a8e77fb4 0x1a8e767a8 0x1a8ee9700 0x1a8ee95f4 0x1a8ee92f0 0x242f7d40c 0x1a8ee917c 0x242f7e498 0x1a914b7ac 0x1a8ff444c 0x1a90b03b8 0x1a90afee0 0x1a90afd0c 0x1bf7cc644 0x1bf7cc4e0 0x1bf7cb10c 0x1bf7d6bf0 0x1023a67bc 0x1023aa1f0 0x1bf7c7490 0x1bf7c7410 0x1bf7c72e8 0x1a6c1d62c 0x1a6c1c8a8 0x1a6c1b0b8 0x1a6c19d88 0x1a6c19968 0x1eaf0f4e0 0x1a908cedc 0x1a908c518 0x1a92c5734 0x10082e028 0x10082dfa0 0x10082e0fc 0x1ca13ad84)
libc++abi: terminating due to uncaught exception of type NSException
```

maybe I have not enabled something in the project settings? 
Also, I enabled run script for install builds only without this checkbox the project would not build with an error that it does not have access.

**Comment by bosborn** [comment]
Try disabling modular headers for PROJ.

`pod 'PROJ', :modular_headers => false`

https://github.com/ngageoint/PROJ/wiki/Usage#module-headers

And you might need `use_modular_headers!` instead of `use_framework!`

**Comment by ecr334** [comment]
My podfile looks like this
```
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'testAppProj' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_modular_headers!
  # Pods for testAppProj

  pod 'proj-ios', :git => 'https://github.com/ngageoint/projections-ios.git', :branch => 'master'
  pod 'PROJ', :modular_headers => false
  pod 'libtiff', :modular_headers => false

end
```
after all manipulations the following error still appears

```
*** Terminating app due to uncaught exception 'Resource Not Found', reason: 'Failed to find resource 'PROJ.bundle/proj' of type 'db''
*** First throw call stack:
(0x1a6cd2b28 0x19eb26f78 0x1a6d582a4 0x104f67ef8 0x104f67c6c 0x104f6a100 0x19eb04c3c 0x19eb07528 0x19eb05e7c 0x19eb04fe0 0x19eb000c4 0x104b08118 0x104b08848 0x1a90c130c 0x1a8e77fb4 0x1a8e767a8 0x1a8ee9700 0x1a8ee95f4 0x1a8ee92f0 0x242f7d40c 0x1a8ee917c 0x242f7e498 0x1a914b7ac 0x1a8ff444c 0x1a90b03b8 0x1a90afee0 0x1a90afd0c 0x1bf7cc644 0x1bf7cc4e0 0x1bf7cb10c 0x1bf7d6bf0 0x1066b67bc 0x1066ba1f0 0x1bf7c7490 0x1bf7c7410 0x1bf7c72e8 0x1a6c1d62c 0x1a6c1c8a8 0x1a6c1b0b8 0x1a6c19d88 0x1a6c19968 0x1eaf0f4e0 0x1a908cedc 0x1a908c518 0x1a92c5734 0x104b0a028 0x104b09fa0 0x104b0a0fc 0x1ca13ad84)
libc++abi: terminating due to uncaught exception of type NSException
```

I've added the necessary files

<img width="1403" alt="Снимок экрана 2024-04-26 в 00 50 02" src="https://github.com/ngageoint/simple-features-proj-ios/assets/16240094/d1f01a36-22d3-4ce4-8150-2bcefa00c338">

**Comment by bosborn** [comment]
You shouldn't need to add the files since you aren't using PROJ directly ([proj-ios adds them](https://github.com/ngageoint/projections-ios/blob/9d2d0aeff2f89304b693bc1fd8423ad4105097d3/proj-ios.podspec#L22)).  I don't think adding them again would cause an issue but also not sure.

Here are some of our podfiles as examples (PROJ is through geopackage-ios):
* https://github.com/ngageoint/marlin-ios/blob/main/Marlin/Podfile (Swift, doesn't specify modular)
* https://github.com/ngageoint/mage-ios/blob/develop/Podfile (Obj-C & Swift, specifies modular but removes for PROJ)
* https://github.com/ngageoint/geopackage-mapcache-ios/blob/develop/Podfile (Obj-C & Swift, recently removed use_frameworks!)

With the most recent CocoaPods version, try manually cleaning and building your project.

```
rm Podfile.lock
rm -rf Pods/
rm -rf *.xcworkspace
pod install
open *.xcworkspace
```

Example:
<img width="757" alt="image" src="https://github.com/ngageoint/simple-features-proj-ios/assets/11667761/065e325b-1de4-4e09-bc6d-e5ca4ad6bffd">

<img width="151" alt="image" src="https://github.com/ngageoint/simple-features-proj-ios/assets/11667761/ef24cfe7-c565-4c59-9e1c-f2262182a745">

<img width="210" alt="image" src="https://github.com/ngageoint/simple-features-proj-ios/assets/11667761/4974787b-a86b-4b1d-b568-74d45e06380a">

<img width="434" alt="image" src="https://github.com/ngageoint/simple-features-proj-ios/assets/11667761/8013b398-5deb-4f72-be6a-cac2d29f2e69">
